Sources/SwiftMemcache/MemcachedConnection.swift

```diff
@@ -137,8 +137,8 @@ public actor MemcachedConnection {
     /// Fetch the value for a key from the Memcache server.
     ///
     /// - Parameter key: The key to fetch the value for.
-    /// - Returns: A `ByteBuffer` containing the fetched value, or `nil` if no value was found.
-    public func get(_ key: String) async throws -> ByteBuffer? {
+    /// - Returns: A `Value` containing the fetched value, or `nil` if no value was found.
+    public func get<Value: MemcachedValue>(_ key: String) async throws -> Value? {
```

Comment:
Not for this PR, but we probably want to create something like a `MemcacheItem<Value: MemcacheValue>` in the future that we return here. This way we could attach metadata such as the TTL or other opaque flags that the user passed.

Reply:
We should also change this method slightly to make it a bit more ergonomic to spell
```suggestion
    /// - Returns: A `Value` containing the fetched value, or `nil` if no value was found.
    public func get<Value: MemcachedValue>(_ key: String, as valueType: Value.Type = Value.self) async throws -> Value? {
```